UserOnboard/Onboard-UserMain.ps1

```diff
@@ -0,0 +1,260 @@
+$customAttributes = @{
+    customAttr1 = 'value1'
+    customAttr2 = 'value2'
+}
+$newADUser = .\Create-ADUser.ps1 @newUserProperties @boundProperties -CustomAttributes $customAttributes
+##
+
+#Add user to ProjectX groups
+$ProjectXGroups = @(
+    'ProjectX_Security'
+    'ProjectX_EMail'
+)
+if ($ProjectX) {
+    Write-Verbose "Adding user to ProjectX groups." -Verbose:$CustomVerbosePreference
+    foreach ($group in $ProjectXGroups) {
+        Write-Verbose "Adding user to group '$group'" -Verbose:$CustomVerbosePreference
+        .\Add-UserToADGroup.ps1 -SamAccountName $SamAccountName -GroupName $group
+    }
+}
+##
+
+#All new users are added to these groups
+$allGroups = @(
+    'security_all',
+    'vpn_access',
+    'distribution_all'
+)
+Write-Verbose "Adding user to groups." -Verbose:$CustomVerbosePreference
+foreach ($group in $allGroups) {
+    Write-Verbose "Adding user to group '$group'" -Verbose:$CustomVerbosePreference
+    .\Add-UserToADGroup.ps1 -SamAccountName $SamAccountName -GroupName $group
+}
+
+
+#Copy template employee groups
+if ($TemplateUser) {
+    Write-Verbose "Copy template user groups." -Verbose:$CustomVerbosePreference
+    $ldapFilter = "(UserPrincipalName=$($TemplateUser.UPN))"
+    $substituteAdUser = Get-ADUser -LDAPFilter $ldapFilter -Properties MemberOf
+    if (-not $substituteAdUser) {
+        Throw "Unable to retrieve substitute user AD Account using filter '$ldapFilter'"
+    }
+    foreach ($group in $substituteAdUser.MemberOf) {
+        .\Add-UserToADGroup.ps1 -SamAccountName $SamAccountName -GroupName $group
+    }    
+}
+##
+
+#Configure Shares
+Write-Verbose "Assigning permissions." -Verbose:$CustomVerbosePreference
+$profileFolder = "\\filesrv\profiles$\$SamAccountName"
+if (-not (Test-Path $profileFolder)) {
+    New-Item -ItemType Directory -Path $profileFolder -Force
+}
+$homefolder = "\\filesrv\users$\$SamAccountName"
+if (-not (Test-Path $homefolder)) {
+    New-Item -ItemType Directory -Path $homefolder -Force
+}
+$systemUserSID = 'S-1-5-18'
+$domainAdminsSID = '' #Fill in your domain admins SID
+Add-NTFSAccess -Path $profileFolder -Account $systemUserSID -AccessRights FullControl
+Add-NTFSAccess -Path $profileFolder -Account $domainAdminsSID -AccessRights FullControl
+Add-NTFSAccess -Path $profileFolder -Account $newADUser.SID -AccessRights Read, ReadAndExecute, Modify
+Disable-NTFSAccessInheritance $profileFolder -RemoveInheritedAccessRules
+Add-NTFSAccess -Path $homefolder -Account $systemUserSID -AccessRights FullControl
+Add-NTFSAccess -Path $homefolder -Account $domainAdminsSID -AccessRights FullControl
+Add-NTFSAccess -Path $homefolder -Account $newADUser.SID -AccessRights Read, ReadAndExecute, Modify
+Disable-NTFSAccessInheritance $homefolder -RemoveInheritedAccessRules
+
+#Configure Exchange
+Write-Verbose "Configure Exchange Account" -Verbose:$CustomVerbosePreference
+$ExchangeServerUri = 'https://exchange.domain.local:443/PowerShell/' #Replace with your Exchange URI
```

Comment:
this should be parameterized if changes from client to client. no hardcoded accepted